src/libstd/path.rs

```diff
@@ -301,7 +302,8 @@ fn os_str_as_u8_slice(s: &OsStr) -> &[u8] {
     unsafe { &*(s as *const OsStr as *const [u8]) }
 }
 unsafe fn u8_slice_as_os_str(s: &[u8]) -> &OsStr {
-    &*(s as *const [u8] as *const OsStr)
+    // SAFETY: see the comment of `os_str_as_u8_slice`
```

Comment:
Hm, so this seems "true" in some sense, but it would be great to figure out why this function is unsafe (whereas the other direction is not) and document that on the function.